contracts/src/v0.8/functions/tests/1_0_0/FunctionsSubscriptions.t.sol

```diff
@@ -1,105 +1,422 @@
+contract FunctionsSubscriptions_GetSubscriptionCount is FunctionsSubscriptionSetup {
+  function test_GetSubscriptionCount_Success() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
+    uint96 subscriptionCount = s_functionsRouter.getSubscriptionCount();
+    // One subscription was made during setup
+    assertEq(subscriptionCount, 1);
+  }
 }
 
 /// @notice #getSubscription
-contract FunctionsSubscriptions_GetSubscription {
+contract FunctionsSubscriptions_GetSubscription is FunctionsSubscriptionSetup {
+  function test_GetSubscription_Success() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    FunctionsSubscriptions.Subscription memory subscription = s_functionsRouter.getSubscription(s_subscriptionId);
 
+    assertEq(subscription.balance, s_subscriptionInitialFunding);
+    assertEq(subscription.owner, OWNER_ADDRESS);
+    assertEq(subscription.blockedBalance, 0);
+    assertEq(subscription.proposedOwner, address(0));
+    assertEq(subscription.consumers[0], address(s_functionsClient));
+    assertEq(subscription.flags, bytes32(0));
+  }
 }
 
 /// @notice #getConsumer
-contract FunctionsSubscriptions_GetConsumer {
+contract FunctionsSubscriptions_GetConsumer is FunctionsSubscriptionSetup {
+  function test_GetConsumer_Success() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
+    FunctionsSubscriptions.Consumer memory consumer = s_functionsRouter.getConsumer(
+      address(s_functionsClient),
+      s_subscriptionId
+    );
+
+    assertEq(consumer.allowed, true);
+    assertEq(consumer.initiatedRequests, 0);
+    assertEq(consumer.completedRequests, 0);
+  }
 }
 
 /// @notice #_isExistingSubscription
-contract FunctionsSubscriptions__IsExistingSubscription {
-
+contract FunctionsSubscriptions__IsExistingSubscription is FunctionsSubscriptionSetup {
+  // TODO: make contract internal function helper
 }
 
 /// @notice #_isAllowedConsumer
 contract FunctionsSubscriptions__IsAllowedConsumer {
-
+  // TODO: make contract internal function helper
 }
 
 /// @notice #createSubscription
-contract FunctionsSubscriptions_createSubscription is FunctionsOwnerAcceptTermsOfService {
-  function setUp() public virtual override {
-    FunctionsOwnerAcceptTermsOfService.setUp();
-  }
-
+contract FunctionsSubscriptions_createSubscription is FunctionsOwnerAcceptTermsOfServiceSetup {
   event SubscriptionCreated(uint64 indexed subscriptionId, address owner);
 
   function test_CreateSubscription_Success() public {
-    vm.expectEmit(true, false, false, true);
+    // topic0 (function signature, always checked), topic1 (true), NOT topic2 (false), NOT topic3 (false), and data (true).
+    bool checkTopic1 = true;
+    bool checkTopic2 = false;
+    bool checkTopic3 = false;
+    bool checkData = true;
+
+    vm.expectEmit(checkTopic1, checkTopic2, checkTopic3, checkData);
```

Comment:
topic 0 = the function signature

the next 3 args are for if there are any ["indexed"](https://docs.soliditylang.org/en/v0.8.19/cheatsheet.html#modifiers) types in the event, which store as a topic rather than event data

the 4th argument to `expectEmit` is to check everything that isn't an indexed type

Reply:
Thanks!